web/yo/app/scripts/services/drugmaputils.spec.js

```diff
@@ -0,0 +1,92 @@
+describe('drugMapUtils', function(){
+    beforeEach(module('oncokbApp'));
+    var drugMapUtilsFactory = null;
+    beforeEach(inject(function(drugMapUtils) {
+        drugMapUtilsFactory = drugMapUtils;
+    }));
+    var drugListExample = {
+        "001e534f-3e63-432f-90a6-d1af1759e4e2" : {
+            "description" : "",
+            "drugName" : "Encorafenib",
+            "ncitCode" : "C98283",
+            "ncitName" : "Encorafenib",
+            "synonyms" : [ "LGX-818", "LGX818", "LGX 818", "ENCORAFENIB", "Braftovi" ],
+            "uuid" : "001e534f-3e63-432f-90a6-d1af1759e4e2"
+        },
+        "0746dd92-8f4d-4bf7-9161-57a223cb67d5" : {
+            "description" : "",
+            "drugName" : "Pembrolizumab",
+            "ncitCode" : "C106432",
+            "ncitName" : "Pembrolizumab",
+            "synonyms" : [ "MK-3475", "PEMBROLIZUMAB", "Keytruda", "Lambrolizumab", "SCH 900475", "Immunoglobulin G4, Anti-(Human Programmed Cell Death 1); Humanized Mouse Monoclonal (228-L-proline(H10-S&gt;P))gamma 4 Heavy Chain (134-218&apos;)-disulfide with Humanized Mouse Monoclonal Kappa Light Chain Dimer (226-226&apos;&apos;:229-229&apos;&apos;)-bisdisulfide" ],
+            "uuid" : "0746dd92-8f4d-4bf7-9161-57a223cb67d5"
+        },
+        "0ade496c-77e6-45cd-8e96-7c7d87b4e5de" : {
+            "description" : "",
+            "drugName" : "MDM2 Antagonist RO5045337",
+            "ncitCode" : "C91724",
+            "ncitName" : "MDM2 Antagonist RO5045337",
+            "synonyms" : [ "R7112", "RO-5045337", "RO5045337" ],
+            "uuid" : "0ade496c-77e6-45cd-8e96-7c7d87b4e5de"
+        },
+        "0f991d49-4cf2-4975-b52f-d7d037aa7f11" : {
+            "description" : "",
+            "drugName" : "Nilotinib",
+            "ncitCode" : "C48375",
+            "ncitName" : "Nilotinib",
+            "synonyms" : [ "AMN 107 Base Form", "nilotinib", "4-Methyl-3-((4-(3-pyridinyl)-2-pyrimidinyl)amino)-N-(5-(4-methyl-1H-imidazol-1-yl)-3-(trifluoromethyl)phenyl)benzamide", "NILOTINIB" ],
+            "uuid" : "0f991d49-4cf2-4975-b52f-d7d037aa7f11"
+        },
+        "122b8921-3cd6-4b57-bdc8-5d39cc5465a1" : {
+            "description" : "",
+            "drugName" : "Ponatinib",
+            "ncitCode" : "C95777",
+            "ncitName" : "Ponatinib",
+            "synonyms" : [ "PONATINIB", "AP-24534", "AP24534", "Benzamide, 3-(2-Imidazo(1,2-B)Pyridazin-3-Ylethynyl)-4-Methyl-N-(4-((4-Methyl-1- Piperazinyl)Methyl)-3-(Trifluoromethyl)Phenyl)" ],
+            "uuid" : "122b8921-3cd6-4b57-bdc8-5d39cc5465a1"
+        }
+    };
+    it('therapyStrToArr should change therapyName from String to Array', function () {
+        expect(drugMapUtilsFactory.therapyStrToArr('Encorafenib + Pembrolizumab, MDM2 Antagonist RO5045337, MDM2 Antagonist RO5045337 + Nilotinib, Ponatinib')).toEqual([['Encorafenib', 'Pembrolizumab'], ['MDM2 Antagonist RO5045337'], ['MDM2 Antagonist RO5045337', 'Nilotinib'], ['Ponatinib']]);
```

Comment:
Could you also do some extreme tests, like  if the input is null, undefined or empt string? You could also try to test some format cases. For instance `Encorafenib +Pembrolizumab`, `Encorafenib+ Pembrolizumab`, ` Encorafenib+Pembrolizumab ` is supposed to be `([['Encorafenib', 'Pembrolizumab']]`

Reply:
For therapyStrToArray function, the input can't be null or undefined. It has been checked before triggering this function. And removing duplicates is not needed and supported here. For other functions, some extreme tests have been included.

Reply:
You don't know where it's going to call this function, so the input is not always guaranteed. If we have typescript, these do not need to be tested. But for now, let's have some test coverages.